packages/lib-web/lib/src/conv.rs

```diff
@@ -0,0 +1,177 @@
+        replace_placeholder_with_blob(root, path, bytes);
+    }
+}
+
+fn replace_placeholder_with_blob(root: &JsValue, path: &[String], bytes: &[u8]) {
+    match path {
+        [head] => {
+            let key = head.into();
+            let blob = make_blob(bytes);
+            unsafe { Reflect::set(&root, &key, &blob) }.unwrap();
+        }
+        [head, ..] => {
+            let key = head.into();
+            let child = unsafe { Reflect::get(&root, &key) }.unwrap();
+            replace_placeholder_with_blob(&child, &path[1..], bytes);
+        }
+        [] => panic!("Empty path"),
+    }
+}
+
+pub fn parse(s: &str, return_blob: bool) -> JsValue {
+    let mut value = json_str_to_serde_value(s);
+    let blobs = if return_blob {
+        replace_strings_with_placeholders(&mut value)
+    } else {
+        vec![]
+    };
+    let jsvalue = serde_to_js_value(&value);
+    replace_placeholders_with_blobs(&jsvalue, &blobs);
+    jsvalue
+}
+
+fn replace_array_buffers_with_placeholders(jsvalue: &JsValue) -> Blobs {
+    let mut blobs: Blobs = vec![];
+    replace_array_buffers_with_placeholders_recursive(jsvalue, &mut blobs, &vec![]);
+    blobs
+}
+
+fn replace_array_buffers_with_placeholders_recursive(
+    jsvalue: &JsValue,
+    blobs: &mut Blobs,
+    path: &Path,
+) {
+    if jsvalue.is_instance_of::<ArrayBuffer>() {
+        // NOTE: don't need to replace ArrayBuffers with placeholders here
+        // because `serde_wasm_bindgen::from_value` will automatically convert them to `{}`
+        let bytes = read_array_buffer(&jsvalue);
+        blobs.push((path.clone(), bytes));
+    } else if jsvalue.is_object() {
+        let obj = Object::from(jsvalue.clone());
+        let keys: js_sys::Array = Object::keys(&obj);
+        for key in keys.to_vec() {
+            let child = unsafe { Reflect::get(&obj, &key) }.unwrap();
+            let mut newpath = path.clone();
+            newpath.push(key.as_string().unwrap());
+            replace_array_buffers_with_placeholders_recursive(&child, blobs, &newpath);
+        }
+    }
+}
+
+fn replace_placeholders_with_strings(root: &mut Value, blobs: &Blobs) {
+    for (path, bytes) in blobs.iter() {
+        replace_placeholder_with_string(root, path, bytes);
+    }
+}
+
+fn replace_placeholder_with_string(root: &mut Value, path: &[String], bytes: &[u8]) {
+    match path {
+        [] => {
+            *root = base64::encode(bytes).into();
+        }
+        [key, ..] => {
+            let obj = root.as_object_mut().unwrap();
+            let child = obj.get_mut(key).unwrap();
+            replace_placeholder_with_string(child, &path[1..], bytes);
+        }
+    }
+}
+
+fn determine_return_blob(obj: &JsValue, blobs: &Blobs) -> bool {
+    // If `response_binary_type` request param is set to 'blob'
+    // or there is any blob in the request params and `response_binary_type` is not set,
+    // then all strings in the response params will be converted from base64 to raw binary JS Blobs
```

Comment:
Can you add that this behaviour is to be changed later, as we hope that SDK core might start sending `binary` types in response?

Reply:
Done in 164fa74c08c49fc2f8ee54f024b8f05835d5f679